### ISSUE TITLE
getDefaultConfigSources do not use given ClassLoader

### ISSUE THREAD

**Issue by mazze5** [microbean/microbean-microprofile-config#3]
Hi,

I use microbean-microprofile-config for the first time and I run into the following issue. I am not sure if this is a bug or if I had to solve it in another way.

In my application I use `ConfigProvider.getConfig(myClassLoader)` to load the config. I use my own ClassLoader, in this ClassLoader the file "META-INF/microprofile-config.properties" exists. But the properties file was not found.
So I debug that issue, following the call-stack
```
ConfigProvider.getConfig(myClassLoader)
	ConfigProviderResolver.instance().getConfig(cl)
		ConfigBuild.build()
			org.microbean.microprofile.config.Config.getDefaultConfigSources()
				Collection<? extends ConfigSource> microprofileConfigPropertiesConfigSources = getMicroprofileConfigPropertiesSources((ClassLoader)null);
					getMicroprofileConfigPropertiesSources(ClassLoader classLoader)
						classLoader.getResources("META-INF/microprofile-config.properties");
```

The properties file will not be found because for loading properties file the default classloader is use at the following line you see that classloader is passed as null to "getMicroprofileConfigPropertiesSources"
https://github.com/microbean/microbean-microprofile-config/blob/4d04dd8ee9cc77aeb10cb7295800d72ddf914821/src/main/java/org/microbean/microprofile/config/Config.java#L674

In my opinion thats wrong. I think method "getDefaultConfigSources" should be overloaded with a parameter "classloader" and the call from ConfigBuild.build() should pass the classloader

**Comment by mazze5** [comment]
I have created a pull request with the described solution. This works fine for my issue.
It would be great if you can merge this and make a new patch release

**Comment by ljnelson** [comment]
I am sincerely sorry for the delay—somehow I never saw this issue (I have a lot of projects).

One thing I have to check is what the rather bad MicroProfile Config specification says about classloaders here.  What you are proposing is entirely reasonable but I'm not sure offhand if the specification permits it.  `ClassLoader`s in MicroProfile Config are, sadly, used not for their classloading abilities, but as kind of markers standing in for applications.  So when you say `ConfigProvider.getConfig(someClassLoader)`, your configuration is registered with that classloader, true, but I don't think there's anything about whether that classloader should be used for actually loading resources.  It has been a while since I've looked at this project or the specification, so kindly bear with me for a little while longer.

**Comment by ljnelson** [comment]
Related: https://github.com/eclipse/microprofile-config/issues/426

**Comment by mazze-5** [comment]
@ljnelson Had you time to check what the MicroProfile Config specification says?

**Comment by ljnelson** [comment]
I have and I think given how loose the specification is your change is reasonable. I will approve it shortly.

**Comment by ljnelson** [comment]
Fix applied and version 0.4.2 released to Maven Central.
